extension/basicauthextension/extension.go

```diff
@@ -33,18 +33,14 @@ type basicAuth struct {
 	matchFunc  func(username, password string) bool
 }
 
-func newClientAuthExtension(cfg *Config) (auth.Client, error) {
-	if cfg.ClientAuth == nil || cfg.ClientAuth.Username == "" {
-		return nil, errNoCredentialSource
-	}
-
```

Comment:
Looks like Config.Validate misses the `ClientAuth.Username == ""` check

Reply:
That check is actually invalid according to RFC 2617. I was a bit confused so I looked it up:
https://datatracker.ietf.org/doc/html/rfc2617#section-2
As described in this SO answer:
https://stackoverflow.com/a/74658311/407903